source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst

```diff
@@ -88,137 +110,290 @@ Also, make sure to have a text editor setup to write Rust.
+       chain.create_account(account_owner);
+       chain.create_account(account_other);
+   }
 
-Now you can call ``piggy_init`` and get a result containing the initial state.
+The balance of the accounts matter when using the testing library, as the cost of transactions, for example deploying a smart contract, will be deducted from the balance of the account sending the transaction.
+
+
+With the accounts created, you are ready to load and deploy the smart contract module.
+|concordium-smart-contract-testing|_ uses the compiled smart contract modules that you also deploy on the block chain.
+Use ``cargo concordium``, which you installed in preparation for this tutorial, to compile the piggy bank to web assembly (wasm).
+
+Open a terminal and use ``cd``, short for *change directory*, to go into the root of your piggy bank project.
+Then compile your contract with:
+
+.. code-block:: console
+
+   $cargo concordium build --out piggy_bank_part2.wasm.v1
+
+This produces the wasm module ``piggy_bank_part2.wasm.v1`` in the root of your project.
+That is, unless you have any typos or bugs in your code.
+If that is the case, try to fix them using the helpful error messages from the compiler or go back to the end of part 1 and copy the full contract code again.
+
+Going back to your test case, use the function |module_load_v1|_ to load the module.
 
 .. code-block:: rust
 
-   let state_result = piggy_init(&ctx, &mut state_builder);
+   // .. Imports and constants omitted for brevity
 
-First, you want the test to fail if the contract did not result in an
-initial state:
+   #[test]
+   fn test_init() {
+       // .. lines omitted for brevity.
+       let module = load_module_v1("piggy_bank_part2.wasm.v1").expect("Module is valid and exists");
+   }
+
+|module_load_v1|_ loads a module on disk, which might be missing or invalid, and it thus returns a ``Result`` type.
+You can use ``unwrap``, or ``expect`` (which you provide a helpful error message), to extract the actual module from the ``Result``.
+Both methods will panic if the ``Result`` actually contains the ``Err`` variant, which in turn will make the test case fail.
+
+Next step is to deploy the module to the ``chain`` with the method |Chain_module_deploy|_.
+
+Since this is a transaction, you must provide an account address of the ``sender``, which will pay for the cost of the transaction.
+You must also provide a |Signer|_ with a number of keys.
+This mimics the behaviour on the real chain, where one or more keys must sign a transaction.
+The only observable difference between using one or more keys is the cost of the transaction, where each extra key increases the cost slightly.
+In this tutorial, you will always use a |Signer|_ with one key as that is the most common scenario.
 
 .. code-block:: rust
 
-       let state = state_result.expect("Contract initialization results in error.");
+   // .. Imports and constants omitted for brevity
+
+   #[test]
+   fn test_init() {
```

Comment:
Maybe we should promote tests with `anyhow::Result` or another Result<_, > type. Then it's very convenient to just put `?` everywhere and you get decent results instead of having to write "expect" everywhere.